src/webui/api/apicontroller.cpp

```diff
@@ -89,3 +90,24 @@ void APIController::setResult(const QJsonObject &result)
 {
     m_result = QJsonDocument(result);
 }
+
+void APIController::setErrors(const QString &errors)
```

Comment:
IMO, these methods are incorrect. Result of API method can contain not only "errors" field. You should build result object inside the method and call `setResult()` at the end. Otherwise some other dev can be confused when it call `setErrors()` and its result will be overwritten.
```c++
setResult(QJsonObject {"errors", errors});
```

Reply:
I think we should merge this PR as it was previously, with less error handling. We can solve the larger Web API error handling problem for all APIs in a separate issue/PR. We shouldn't jam up feature development because of it.

Reply:
>I think we should merge this PR as it was previously, with less error handling.

:+1:
We can merge it without the last commit for now.
>We can solve the larger Web API error handling problem for all APIs in a separate issue/PR.

Since error handling in entire Web API is problem itself it should be discussed and solved separately (to not produce yet another inconsistencies).

@Chocobo1, what do you say?